.github/workflows/build-integration.yml

```diff
@@ -0,0 +1,51 @@
+name: gocardless/stolon/build-integration
+on:
+  push:
+    branches:
+    - gocardless-v0.17.0+
+env:
+  GITHUB_TOKEN: xxxx6d08
+jobs:
+  unit-tests:
+    defaults:
+      run:
+        working-directory: "/go/src/github.com/sorintlab/stolon"
+    runs-on: ubuntu-latest
+    container:
+      image: golang:1.16
+    steps:
+    - uses: actions/checkout@v2
+    - name: Unit tests
+      run: make test
+  release:
+    if: contains('refs/heads/gocardless-v0.17.0+', github.ref)
+    defaults:
+      run:
+        working-directory: "/go/src/github.com/sorintlab/stolon"
+    runs-on: ubuntu-latest
+    container:
+      image: golang:1.16
+    needs:
+    - unit-tests
+    steps:
+    - uses: actions/checkout@v2
+    # Ensure parameter if_key_exists is set correctly
+    - name: Install SSH key
+      uses: shimataro/ssh-key-action@v2
+      with:
+        key: "${{ secrets.CIRCLE_CI_SSH_KEY }}"
+        name: circle_ci_id_rsa
+        known_hosts: "${{ secrets.CIRCLE_CI_KNOWN_HOSTS }}"
+        if_key_exists: fail
+    - name: Release
+      run: |-
+        CURRENT_VERSION="v$(cat VERSION)"
+        if [[ $(git tag -l "${CURRENT_VERSION}") == "${CURRENT_VERSION}" ]]; then
+          echo "Version ${CURRENT_VERSION} is already released"
+          exit 0
+        fi
+        curl -L -o /tmp/goreleaser_Linux_x86_64.tar.gz https://github.com/goreleaser/goreleaser/releases/download/v1.7.0/goreleaser_Linux_x86_64.tar.gz
+        tar zxf /tmp/goreleaser_Linux_x86_64.tar.gz -C /tmp
+        git tag "${CURRENT_VERSION}"
+        git push --tags
+        /tmp/goreleaser --rm-dist
```

Comment:
We will want to use workload identity federation to push to gcr here. See our [docs](https://backstage.gocardless.io/docs/default/component/github-actions/CI/03-authentication/#authenticating-with-workload-identity-federation) for further info and the [docker/login-action](https://github.com/docker/login-action#google-container-registry-gcr) for how this will look.

It isn't clear to me that we'll need to add an ssh key, but if we do, see the relevant bit in our docs [here](https://backstage.gocardless.io/docs/default/component/github-actions/CI/03-authentication/#ssh-keys-and-accessing-private-repositories)